File: internal/store/ids.go
package store

const (
	sessionPrefix  = "sess-"
	resourcePrefix = "sesrsc-"
	eventPrefix    = "evt-"
	schedulePrefix = "sched-"
	teamPrefix     = "team-"
	memberPrefix   = "tmem-"
	tmsgPrefix     = "tmsg-"
	ttaskPrefix    = "ttsk-"
	threadPrefix   = "sthr_"
)

func generateSessionID() string {
	return sessionPrefix + randomString(idLength)
}

func generateResourceID() string {
	return resourcePrefix + randomString(idLength)
}

// NewResourceID returns a new session resource identifier.
func NewResourceID() string {
	return generateResourceID()
}

func generateEventID() string {
	return eventPrefix + randomString(idLength)
}

func generateScheduleID() string {
	return schedulePrefix + randomString(idLength)
}

func generateTeamID() string {
	return teamPrefix + randomString(idLength)
}

func generateTeamMemberID() string {
	return memberPrefix + randomString(idLength)
}

func generateTeamMessageID() string {
	return tmsgPrefix + randomString(idLength)
}

func generateTeamTaskID() string {
	return ttaskPrefix + randomString(idLength)
}

func generateThreadID() string {
	return threadPrefix + randomString(idLength)
}

// NewEventID returns a unique event id for span / lifecycle events.
func NewEventID() string {
	return generateEventID()
}

// NewScheduleID returns a unique wakeup schedule id.
func NewScheduleID() string {
	return generateScheduleID()
}

// NewTeamID returns a unique team id.
func NewTeamID() string {
	return generateTeamID()
}

// NewTeamMemberID returns a unique team member id.
func NewTeamMemberID() string {
	return generateTeamMemberID()
}

// NewTeamMessageID returns a unique team mailbox message id.
func NewTeamMessageID() string {
	return generateTeamMessageID()
}

// NewTeamTaskID returns a unique team task id.
func NewTeamTaskID() string {
	return generateTeamTaskID()
}

// NewThreadID returns a unique session thread id (non-primary).
func NewThreadID() string {
	return generateThreadID()
}

File: harness/pi_team/ids.py
"""ID generation aligned with internal/store/ids.go."""

from __future__ import annotations

import secrets
import string

_ID_ALPHABET = string.digits + string.ascii_lowercase
_ID_LENGTH = 16

PRIMARY_THREAD_ID = "sthr_primary"


def _random_string(length: int = _ID_LENGTH) -> str:
    return "".join(secrets.choice(_ID_ALPHABET) for _ in range(length))


def new_team_id() -> str:
    return f"team-{_random_string()}"


def new_team_member_id() -> str:
    return f"tmem-{_random_string()}"


def new_team_message_id() -> str:
    return f"tmsg-{_random_string()}"


def new_thread_id() -> str:
    return f"sthr_{_random_string()}"
